backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/StoredServerInfo.java

```diff
@@ -19,22 +19,35 @@
  */
 package org.sonarsource.sonarlint.core.serverconnection;
 
+import javax.annotation.Nullable;
 import org.sonarsource.sonarlint.core.commons.Version;
 
 public class StoredServerInfo {
+
   private final Version version;
-  private final boolean isMQRMode;
+  private final ModeDetails isMQRMode;
 
-  public StoredServerInfo(Version version, boolean isMQRMode) {
+  public StoredServerInfo(Version version, @Nullable Boolean mode) {
     this.version = version;
-    this.isMQRMode = isMQRMode;
+    if (mode == null) {
+      this.isMQRMode = ModeDetails.DEFAULT;
+    } else if (mode) {
+      this.isMQRMode = ModeDetails.MQR;
+    } else {
+      this.isMQRMode = ModeDetails.STANDARD;
+    }
   }
 
   public Version getVersion() {
     return version;
   }
 
-  public boolean isMQRMode() {
+  public ModeDetails isMQRMode() {
```

Comment:
I find it strange that a `is` method does not return a boolean, maybe worth renaming the method since you rely on an enum?

Reply:
Ah yes, you're right, I wanted to modify this one because there is confusion with the SeverityModeService that returns the boolean